pkg/cri/server/sandbox_portforward_linux.go

```diff
@@ -62,12 +62,24 @@ func (c *criService) portForward(ctx context.Context, id string, port int32, str
 	log.G(ctx).Infof("Executing port forwarding in network namespace %q", netNSPath)
 	err = netNSDo(func(_ ns.NetNS) error {
 		defer stream.Close()
-		// TODO: hardcoded to tcp4 because localhost resolves to ::1 by default if the system has IPv6 enabled.
-		// Theoretically happy eyeballs will try IPv6 first and fallback to IPv4
-		// but resolving localhost doesn't seem to return and IPv4 address, thus failing the connection.
```

Comment:
this is wrong, we assumed that the system resolver wasn't working or the net.Dial was not falling back, but the problem was that the connection was created outside of the namespace, and there was nothing listening on localhost there